darts/models/forecasting/baselines.py

```diff
@@ -215,51 +331,44 @@ def __init__(self, input_chunk_length: int = 1):
         """
         super().__init__()
         self.input_chunk_length = input_chunk_length
-        self.rolling_window = None
 
-    @property
-    def supports_multivariate(self) -> bool:
-        return True
+    def _fit_model(self, series: Union[TimeSeries, Sequence[TimeSeries]]) -> None:
+        super()._fit_model(series)
 
-    @property
-    def min_train_series_length(self):
-        return self.input_chunk_length
+    def _predict(
+        self, n: int, series: Sequence[TimeSeries] = None, num_samples: int = 1
+    ) -> Sequence[TimeSeries]:
 
-    def __str__(self):
-        return f"NaiveMovingAverage({self.input_chunk_length})"
+        super()._predict(n, num_samples)
 
-    def fit(self, series: TimeSeries):
-        super().fit(series)
-        raise_if_not(
-            series.is_deterministic,
-            "This model expects deterministic time series",
-            logger,
-        )
+        series = series2seq(series)
+        predictions = []
 
-        self.rolling_window = series[-self.input_chunk_length :].values(copy=False)
-        return self
-
-    def predict(
-        self,
-        n: int,
-        num_samples: int = 1,
-        verbose: bool = False,
-    ):
-        super().predict(n, num_samples)
-
-        predictions_with_observations = np.concatenate(
-            (self.rolling_window, np.zeros(shape=(n, self.rolling_window.shape[1]))),
-            axis=0,
-        )
-        rolling_sum = sum(self.rolling_window)
-
-        chunk_length = self.input_chunk_length
-        for i in range(chunk_length, chunk_length + n):
-            prediction = rolling_sum / chunk_length
-            predictions_with_observations[i] = prediction
-            lost_value = predictions_with_observations[i - chunk_length]
-            rolling_sum += prediction - lost_value
-        return self._build_forecast_series(predictions_with_observations[-n:])
+        for series_ in series:
+            rolling_window = series_[-self.input_chunk_length :].values(copy=False)
+            predictions_with_observations = np.concatenate(
+                (
+                    rolling_window,
+                    np.zeros(shape=(n, rolling_window.shape[1])),
+                ),
+                axis=0,
+            )
+            rolling_sum = sum(rolling_window)
+
+            chunk_length = self.input_chunk_length
+            for i in range(chunk_length, chunk_length + n):
+                prediction = rolling_sum / chunk_length
```

Comment:
this is a constant. You can put it outside of the for loop

or it is an error?